### PR TITLE
use dep instead of glide and simplify the build process

### PR DESCRIPTION
**Pull request by xcoulon**
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
[test]

File: Makefile [hunk 1]
```diff
@@ -1,174 +1,171 @@
...
 
-# For the global "clean" target all targets in this variable will be executed
-CLEAN_TARGETS =
+.PHONY: build-linux
+build-linux: vendor generate $(BUILD_DIR) # Builds the Linux binary for the container image into $BUILD_DIR
+	CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -v $(LDFLAGS) -o $(BUILD_DIR)/$(REGISTRY_IMAGE)
 
-# Pass in build time variables to main
-LDFLAGS=-ldflags "-X ${PACKAGE_NAME}/controller.Commit=${COMMIT} -X ${PACKAGE_NAME}/controller.BuildTime=${BUILD_TIME}"
+image: clean-artifacts $(BUILD_DIR)/$(REGISTRY_IMAGE) ## Builds the container image using the binary compiled for Linux
```
Comment (corinnekrych) [inline Makefile]: $(BUILD_DIR)/$(REGISTRY_IMAGE) has been rename to `build` therefore the CentOS CI error `make: *** No rule to make target `bin/fabric8-toggles-service', needed by `image'.  Stop.`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# Docker file to copy the generated binary from the `out` directory
+# -----------------------------------------------------------------
+FROM centos:7
+LABEL author Xavier Coulon <xcoulon@redhat.com>
+
+EXPOSE 8080
+ARG BINARY
+COPY ${BINARY} /usr/local/bin/fabric8-toggles-service
+
+ENTRYPOINT [ "fabric8-toggles-service" ]
```
Comment (aslakknutsen) [inline Dockerfile]: Have you tried to run this on a secured OS?

**Comment by corinnekrych** [comment]
@xcoulon i've dome dome README cleaning +  minishift target refactor in there: https://github.com/xcoulon/fabric8-toggles-service/pull/2

Overall, I've tested the PR and I ❤️  it very much bc it simplified our Makefile so much!

File: Dockerfile (hunk 2 above)
Comment (xcoulon) [reply Dockerfile]: ah right, I forgot the `USER` directive

Comment (aslakknutsen) [reply Dockerfile]: I think this should work when in /usr/local/bin as long as no disk access etc is attempted.. But not sure. Hench, have you tried? :)

**Pull request merged**